todo/src/main.rs

```diff
@@ -1,31 +1,90 @@
 use crate::terminal::Terminal;
+use controllers::todo::{TodoController, TodoControllerImpl};
 use db::connect::connect;
 use repository::todo::PostgresTodoRepository;
-use todocli::TodoCli;
+use terminal::{error::TerminalError, UserInterface, UserOptions};
+mod controllers;
 mod db;
 mod domain;
 mod repository;
 mod terminal;
-mod todocli;
 extern crate factori;
 
 #[tokio::main]
 async fn main() {
     let client = connect()
         .await
         .expect("Database connection error. Quitting");
+    let todo_repository = Box::new(PostgresTodoRepository { client });
     let user_interface = Box::new(Terminal::new());
-    let todo_storage = Box::new(PostgresTodoRepository { client });
-    let mut todo_cli = TodoCli {
+
+    let mut todo_controller = Box::new(TodoControllerImpl {
+        todo_repository,
         user_interface,
-        todo_storage,
-    };
+    });
+
     loop {
-        if let Err(error) = todo_cli.run().await {
-            todo_cli.user_interface.print_error(&error);
+        if let Err(error) =
+            get_user_intention(&mut Box::new(Terminal::new()), &mut todo_controller).await
+        {
+            println!("{}", &error);
             if error.is_fatal() {
                 break;
             }
         }
     }
 }
+
+async fn get_user_intention(
+    user_interface: &mut Box<Terminal>,
```

Comment:
Embora esses argumentos não serão mais necessários, você _poderia_ ter recebido `&mut Terminal` (ou melhor ainda,  `&mut dyn UserInterface`, já que `Box` implementa [`Deref`](https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-Deref-for-Box%3CT%2C%20A%3E) e [`DerefMut`](https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-DerefMut-for-Box%3CT%2C%20A%3E), o que te permite pegar uma referência ao que está dentro dele.